data-prepper-core/src/test/java/org/opensearch/dataprepper/pipeline/ProcessWorkerTest.java

```diff
@@ -174,8 +174,11 @@ void testProcessWorkerWithProcessorThrowingExceptionAndAcknowledgmentsEnabledIsH
         final Record<Event> mockRecord = mock(Record.class);
         final Event mockEvent = mock(Event.class);
         final EventHandle eventHandle = mock(DefaultEventHandle.class);
-        when(((DefaultEventHandle) eventHandle).getAcknowledgementSet()).thenReturn(mock(AcknowledgementSet.class));
-        doNothing().when(eventHandle).release(true);
+        final AcknowledgementSet acknowledgementSet = mock(AcknowledgementSet.class);
+        lenient().doAnswer(a-> {
```

Comment:
I think this can just be `when(acknowledgementSet).release(eventHandle, true).thenReturn(null);`